core/plugin/zplugin.go

```diff
@@ -56,4 +56,7 @@ import (
 	_ "github.com/coredns/coredns/plugin/tsig"
 	_ "github.com/coredns/coredns/plugin/view"
 	_ "github.com/coredns/coredns/plugin/whoami"
+
+	
```

Comment:
@gcs278 can you please run go generate coredns.go for generating this file? I had added one extra line as a hack for passing the import ordering unit test. As you have fixed the generation of this file we should use the correctly generated file then.

Reply:
Hmm. This is an interesting problem. Yes we could re-generate the file, but I'd have to do that in the cherry-pick commit: https://github.com/openshift/coredns/pull/118/commits/2b42769a2f877c2fbae73d543cb7fe4d4385457b because that's where the fix is in and it comes after this commit for ocp nameresolver.

But that seems problematic when we drop this carry patch when the "fix" is in the next CoreDNS rebase. Then the go generate will disappear...

I think our options are:
1. Re-arrange https://github.com/openshift/coredns/pull/118/commits/2b42769a2f877c2fbae73d543cb7fe4d4385457b  (go generate coredns.go "fix") and https://github.com/openshift/coredns/pull/118/commits/d5789ebac4a2a489d025b10cb261b66c1e04d9f2 (ocp nameresolver) so that the go generate "fix" comes first, and then go a `go generate coredns.go` on the ocp nameresolver test in thiis PR.
2. We don't do anything, CI still passes since `go generate coredns.go` runs before unit tests. This also feels a bit kludgy, since the generated file in the local repo is left incorrectly to be regenerated in CI.
3. We add merge another PR in master that only does a `go generate coredns.go` and re-do this rebase to pull that commit in, and squash it into https://github.com/openshift/coredns/pull/118/commits/d5789ebac4a2a489d025b10cb261b66c1e04d9f2 (ocp nameresolver).

I'm also considering the backport for 4.16 of the "fix": https://github.com/openshift/coredns/pull/123.

I think # 3 is the most "correct" approach because it provides a discrete commit we can backport to 4.16, but it feels like overkill to do all of that. But I think doing # 1, and moving forward with the master branch having the correctly generated `zplugin.go` file, is acceptable. For the 4.16 backport, https://github.com/openshift/coredns/pull/123, we can just leave the 4.16 `zplugin.go` alone. CI will still get fixed in 4.16 as long as https://github.com/openshift/coredns/pull/123 gets merged.

Reply:
Done. Re-Arranged and ran `go generate coredns.go`. PTAL.

Reply:
I agree with you that # 1 is probably the fix we can go with.
I have another PR (https://github.com/openshift/coredns/pull/122) which I need to backport to 4.16. I can probably do a manual backport and re-generate the `zplugin.go` in the backport PR. WDYT?

Reply:
Yea I think that's fine. I would probably still separate it as a separate commit to be organized. And you'll need to wait for https://github.com/openshift/coredns/pull/123 to be backported.

I forgot to mention that the risk with NOT committing the generated "fixed" zplugin file is: the `core/plugin/zplugin.go` will get regenerated locally when you run `make`. So that may confuse people doing 4.16 backports, but it's not actually problematic, just annoying.